### PR TITLE
[DOCS-6755] - Search Enterprise zero downtime upgrade

### PR DESCRIPTION
**Pull request by oliveralfresco2018**

### CODE REVIEW

File: search-enterprise/latest/upgrade/index.md [hunk 1]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
 
 > **Note:** You may need to use the Elasticsearch Re-indexing application to update to the latest changes. After that, new and updated documents will be uploaded to the Elasticsearch index by the Elasticsearch connector service using ActiveMQ messages.
+
+## Zero downtime upgrade
+
+You can upgrade from Search Services 2.x without experiencing any downtime, to Search Enterprise 3.1.1 when you are using Content Services 7.2 and above.
+
+1. Start an Elasticsearch 3.1.1 instance, for more see [Overview]({% link search-enterprise/latest/install/index.md %}).
+
+    Currently your installation currently only using solr.
```
Comment (mjstanford) [inline search-enterprise/latest/upgrade/index.md:53]: Not great English

File: search-enterprise/latest/upgrade/index.md [hunk 2]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+    **Note:** The index is not yet in sync with the primary environment.
+
+    ![shutdown-mirrored]({% link search-enterprise/images/shutdown-mirrored.png %})
+
+6. Keep your index up to date with changes made through Content Services.
+
+    To do this start live indexing on the primary environment. For more see [live-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-live-indexing-app).
+
+    > **Note:** Even after starting live indexing there is still a gap however between when you took a snapshot and when you started live indexing.
```
Comment (mjstanford) [inline search-enterprise/latest/upgrade/index.md:89]: Suggest "Even after starting live indexing there is still a gap from when you took a snapshot to when you started live indexing."

File: search-enterprise/latest/upgrade/index.md (hunk 1 above)
Comment (oliveralfresco2018) [reply search-enterprise/latest/upgrade/index.md]: Changed

File: search-enterprise/latest/upgrade/index.md (hunk 2 above)
Comment (oliveralfresco2018) [reply search-enterprise/latest/upgrade/index.md]: Changed

**Review by mjstanford — APPROVED** [review]

File: search-enterprise/latest/upgrade/index.md [hunk 3]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+    ![start-live-indexing]({% link search-enterprise/images/start-live-indexing.png %})
+
+7. Start re-indexing on the solr environment.
+
+    To close the gap in the Elasticsearch index start re-indexing on the solr environment. **Note:** Live indexing keeps the Elasticsearch environment index up to date.
+
+    ![final-reindexing]({% link search-enterprise/images/final-re-Indexing.png %})
+
+8. Switch to Elasticsearch
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:99]:
```suggestion
8. Switch to Elasticsearch.
```

File: search-enterprise/latest/upgrade/index.md [hunk 4]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+    To switch to Elasticsearch access the Admin Console at runtime. Once you have done this you still have both Search Services and Search Enterprise running but Content Services is using Elasticsearch.
+
+   > **Note:** If you experience any issues you can still revert back to using solr.
+
+    ![switch-elasticsearch]({% link search-enterprise/images/switch-elasticsearch.png %})
+
+9. Shutdown Search Services
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:107]:
```suggestion
9. Shutdown Search Services.
```

File: search-enterprise/latest/upgrade/index.md [hunk 5]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
 
 > **Note:** You may need to use the Elasticsearch Re-indexing application to update to the latest changes. After that, new and updated documents will be uploaded to the Elasticsearch index by the Elasticsearch connector service using ActiveMQ messages.
+
+## Zero downtime upgrade
+
+You can upgrade from Search Services 2.x without experiencing any downtime, to Search Enterprise 3.1.1 when you are using Content Services 7.2 and above.
+
+1. Start an Elasticsearch 3.1.1 instance, for more see [Overview]({% link search-enterprise/latest/install/index.md %}).
+
+    Currently your installation is using solr.
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:53]: Change solr to Solr in this page (there are several instances).

File: search-enterprise/latest/upgrade/index.md [hunk 6]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+    To do this start live indexing on the primary environment. For more see [live-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-live-indexing-app).
+
+    > **Note:** Even after starting live indexing there is still a gap from when you took a snapshot to when you started live indexing.
+
+    ![start-live-indexing]({% link search-enterprise/images/start-live-indexing.png %})
+
+7. Start re-indexing on the solr environment.
+
+    To close the gap in the Elasticsearch index start re-indexing on the solr environment. **Note:** Live indexing keeps the Elasticsearch environment index up to date.
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:95]: Fix the formatting for all notes so they're similar to line 89 (there are several instances).

File: search-enterprise/latest/upgrade/index.md [hunk 7]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+You can upgrade from Search Services 2.x without experiencing any downtime, to Search Enterprise 3.1.1 when you are using Content Services 7.2 and above.
+
+1. Start an Elasticsearch 3.1.1 instance, for more see [Overview]({% link search-enterprise/latest/install/index.md %}).
+
+    Currently your installation is using solr.
+
+    ![add-empty]({% link search-enterprise/images/add-empty-elasticsearch.png %})
+
+2. Start a mirrored environment by replicating the content respository and Content Services.
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:57]: Typo
```suggestion
2. Start a mirrored environment by replicating the content repository and Content Services.
```

File: search-enterprise/latest/upgrade/index.md [hunk 8]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+    > **Note:** The index is not created when you mirrored the content repository and Content Services.
+
+4. Populate the index with existing data.
+
+    The index is populated and is based on the replicated database and is achieved by starting re-indexing on the mirrored environment. For more see [re-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-re-indexing-app).
+
+    **Note:** A window displays that states the primary database does not reflect the up to date index.
+
+    ![intial-reindexing]({% link search-enterprise/images/initial-re-indexing.png %})
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:75]:
```suggestion
    ![initial-reindexing]({% link search-enterprise/images/initial-re-indexing.png %})
```

File: search-enterprise/latest/upgrade/index.md (hunk 5 above)
Comment (oliveralfresco2018) [reply search-enterprise/latest/upgrade/index.md:53]: Updated

**Review by gravitonian — APPROVED** [review]
LGTM. Just some notes that have double > >

File: search-enterprise/latest/upgrade/index.md [hunk 9]
```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
...
+
+    To do this start live indexing on the primary environment. For more see [live-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-live-indexing-app).
+
+    > **Note:** Even after starting live indexing there is still a gap from when you took a snapshot to when you started live indexing.
+
+    ![start-live-indexing]({% link search-enterprise/images/start-live-indexing.png %})
+
+7. Start re-indexing on the Solr environment.
+
+    To close the gap in the Elasticsearch index start re-indexing on the Solr environment. > **Note:** Live indexing keeps the Elasticsearch environment index up to date.
```
Comment (anxumalo) [inline search-enterprise/latest/upgrade/index.md:95]: Format the note as in line 89

**Review by anxumalo — CHANGES_REQUESTED** [review]
Some of your earlier commits that were resolved seem to have been overwritten by later commits. Please check the sections again.

**Review by anxumalo — APPROVED** [review]
LGTM

**Pull request merged**